### PR TITLE
RC-v1.8: Liquid/Lock account dashboard 

### LINKED ISSUE #1758: Liquid/locked account dashboard
**CU Ticket:** https://app.clickup.com/t/865bpuphz
**Assigned To:** Justin
**Version:**

### PR DESCRIPTION
**Pull request by ap-justin**
Ticket(s):
close #1758 

## Explanation of the solution
* create dashboard
NOTE: to see positions (even with `0` invested) should comment out this https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1744/commits/06664b601f8e7ff9921377ea48006060bf7ea582
image

* redeem form and wiring
image

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/pages/Admin/Charity/Account/Balances.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const queryState = useAssetQuery({ id, type });
+
+  return (
+    <QueryLoader
+      queryState={queryState}
+      messages={{ loading: <Skeleton />, error: "Failed to get balances" }}
+    >
+      {({ total, free, invested, symbol }) => (
+        <div className="grid grid-cols-3 gap-6">
+          <Amount title="Total value" symbol={symbol}>
```
Comment (misicnenad) [inline src/pages/Admin/Charity/Account/Balances.tsx:20]: **Suggestion:**

`Amount` usage can rewritten to:
```tsx
          ...
          <Amount title="Total value" value={total} symbol={symbol} />
          <Amount title="Invested balances" value={invested} symbol={symbol} />
          <Amount title="Free balances" value={free} symbol={symbol} />
          ...
          
function Amount({
  classes = "",
  ...props
}: PropsWithChildren<{ classes?: string; title: string; value: number; symbol: string }>) {
  return (
    <div
      className={
        classes +
        " border border-prim py-4 px-5 rounded bg-orange-l6 dark:bg-blue-d6"
      }
    >
      <p className="text-xs text-gray-d1 dark:text-gray mb-2 uppercase">
        {props.title}
      </p>
      <span className="font-bold text-xl font-heading">
        {humanize(props.value, 2)}{" "}
        <span className="text-xs font-normal">{props.symbol}</span>
      </span>
    </div>
  );
}
```

**Review by misicnenad — CHANGES_REQUESTED** [review]

File: src/pages/Admin/Charity/Account/Balances.tsx (hunk 1 above)
Comment (misicnenad) [inline src/pages/Admin/Charity/Account/Balances.tsx:20]: **Suggestion:**

`Amount` usage can rewritten to:
```tsx
          ...
          <Amount title="Total value" value={total} symbol={symbol} />
          <Amount title="Invested balances" value={invested} symbol={symbol} />
          <Amount title="Free balances" value={free} symbol={symbol} />
          ...
          
function Amount({
  classes = "",
  ...props
}: PropsWithChildren<{ classes?: string; title: string; value: number; symbol: string }>) {
  return (
    <div
      className={
        classes +
        " border border-prim py-4 px-5 rounded bg-orange-l6 dark:bg-blue-d6"
      }
    >
      <p className="text-xs text-gray-d1 dark:text-gray mb-2 uppercase">
        {props.title}
      </p>
      <span className="font-bold text-xl font-heading">
        {humanize(props.value, 2)}{" "}
        <span className="text-xs font-normal">{props.symbol}</span>
      </span>
    </div>
  );
}
```

Comment (misicnenad) [inline src/pages/Admin/Charity/Account/Balances.tsx:20]: **Suggestion:**

`Amount` usage can rewritten to:
```tsx
          ...
          <Amount title="Total value" value={total} symbol={symbol} />
          <Amount title="Invested balances" value={invested} symbol={symbol} />
          <Amount title="Free balances" value={free} symbol={symbol} />
          ...
          
function Amount({
  classes = "",
  ...props
}: PropsWithChildren<{ classes?: string; title: string; value: number; symbol: string }>) {
  return (
    <div
      className={
        classes +
        " border border-prim py-4 px-5 rounded bg-orange-l6 dark:bg-blue-d6"
      }
    >
      <p className="text-xs text-gray-d1 dark:text-gray mb-2 uppercase">
        {props.title}
      </p>
      <span className="font-bold text-xl font-heading">
        {humanize(props.value, 2)}{" "}
        <span className="text-xs font-normal">{props.symbol}</span>
      </span>
    </div>
  );
}
```

File: src/pages/Admin/Charity/Account/Balances.tsx [hunk 2]
```diff
@@ -18,24 +17,28 @@ export default function Balances({ type }: { type: AccountType }) {
...
 
-function Amount({
-  classes = "",
-  ...props
-}: PropsWithChildren<{ classes?: string; title: string; symbol: string }>) {
+type Props = {
+  classes?: string;
+  title: string;
+  symbol: string;
+  children: number;
```
Comment (misicnenad) [inline src/pages/Admin/Charity/Account/Balances.tsx:38]: Using `props.children` like this is a bit confusing, would've been better to use the suggested `value: number` (or whatever other name) field.

**Review by misicnenad — APPROVED** [review]

File: src/pages/Admin/Charity/Account/Balances.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      className={
+        classes +
+        " border border-prim py-4 px-5 rounded bg-orange-l6 dark:bg-blue-d6"
+      }
+    >
+      <p className="text-xs text-gray-d1 dark:text-gray mb-2 uppercase">
+        {props.title}
+      </p>
+      <span className="font-bold text-xl font-heading">
+        {humanize(props.children, 2)}{" "}
```
Comment (misicnenad) [inline src/pages/Admin/Charity/Account/Balances.tsx:53]: @ap-justin using `props.children` like this is a bit confusing, would've been better to use the suggested `value: number` (or whatever other name) field, not to mention there would be no need for the closing `</Amount>` tag.

**Review by misicnenad — APPROVED** [review]

File: src/pages/Admin/Charity/Account/Balances.tsx (hunk 3 above)
Comment (ap-justin) [reply src/pages/Admin/Charity/Account/Balances.tsx:53]: I dont find this confusing jsx children is `tyepof ReactNode` which `number` is a member of

Comment (misicnenad) [reply src/pages/Admin/Charity/Account/Balances.tsx:53]: Would still be shorter and clearer with the prop.

**Pull request merged**